### PR TITLE
Add detailed upgrade steps to v10/v11 migration guide

### PR DESCRIPTION
**Pull request by lgriffee**
### WHY are these changes introduced?
We want to try and give folks all the tools we've been using to check the migrations. 

### WHAT is this pull request doing?
* Added post-migration validation checks for all migrations
* Added recommended migration workflows for both component and token sections
    * Added the following sentence to all migrations to point to the recommended workflow guide: “Please reference the `[recommended XX migration workflow](#recommended-XX-migration-workflow)` section below for additional migration support.
* Broke out all component and token sections into additional subsections for clarity (Migration, Post-Migration Validation, Replacement Maps)
* Added backticks to all component names for increased scanability 
* Fixed naming for the Shadow section (Depth ➡️ Shadow)

### Tophatting 🎩
I'd recommend taking a look [here at the file itself](https://github.com/Shopify/polaris/blob/11c27fa4fec74e8ce167d86c27664ddf93de8f39/documentation/guides/migrating-from-v10-to-v11.md) versus the diff.

### CODE REVIEW

File: documentation/guides/migrating-from-v10-to-v11.md [hunk 1]
```diff
@@ -9,32 +9,34 @@ Polaris v11.0.0 ([full release notes](https://github.com/Shopify/polaris/release
 - [Webpack support](#webpack-support)
 - [TypeScript](#type-script)
```
Comment (laurkim) [inline documentation/guides/migrating-from-v10-to-v11.md]:
```suggestion
- [TypeScript](#typescript)
```

I think this was an error from my PR, sorry 😅 checked locally and removing the hyphen fixes the link

File: documentation/guides/migrating-from-v10-to-v11.md [hunk 2]
```diff
@@ -9,32 +9,34 @@ Polaris v11.0.0 ([full release notes](https://github.com/Shopify/polaris/release
...
 - [Components](#components)
-  - [Removed Collapsible deprecated preventMeasuringOnChildrenUpdate prop](#removed-collapsible-deprecated-prevent-measuring-on-children-update-prop)
```
Comment (laurkim) [inline documentation/guides/migrating-from-v10-to-v11.md:12]:
```suggestion
  - [Removed `Collapsible` deprecated `preventMeasuringOnChildrenUpdate` prop](#removed-collapsible-deprecated-preventmeasuringonchildrenupdate-prop)
```

File: documentation/guides/migrating-from-v10-to-v11.md [hunk 3]
```diff
@@ -9,32 +9,34 @@ Polaris v11.0.0 ([full release notes](https://github.com/Shopify/polaris/release
...
-  - [Migrated Stack to LegacyStack](#migrated-stack-to-legacy-stack)
-  - [Migrated Card to LegacyCard](#migrated-card-to-legacy-card)
-  - [Migrated Filters to LegacyFilters](#migrated-filters-to-legacy-filters)
-  - [Migrated Tabs to LegacyTabs](#migrated-tabs-to-legacy-tabs)
-  - [Renamed Inline](#renamed-inline)
-  - [Renamed AlphaStack](#renamed-alpha-stack)
-  - [Renamed AlphaCard](#renamed-alpha-card)
-  - [Renamed AlphaFilters](#renamed-alpha-filters)
-  - [Renamed AlphaTabs](#renamed-alpha-tabs)
+  - [Removed `Collapsible` deprecated `preventMeasuringOnChildrenUpdate` prop](#removed-collapsible-deprecated-prevent-measuring-on-children-update-prop)
```
Comment (laurkim) [inline documentation/guides/migrating-from-v10-to-v11.md]:
```suggestion
  - [Removed `Collapsible` deprecated `preventMeasuringOnChildrenUpdate` prop](#removed-collapsible-deprecated-preventmeasuringonchildrenupdate-prop)
```

Having the prop name in backticks breaks the original url

File: documentation/guides/migrating-from-v10-to-v11.md [hunk 4]
```diff
@@ -9,32 +9,34 @@ Polaris v11.0.0 ([full release notes](https://github.com/Shopify/polaris/release
...
+  - [Removed `Page` deprecated `breadcrumbs` prop](#removed-page-deprecated-breadcrumbs-prop)
+  - [Removed `Breadcrumbs` deprecated `breadcrumbs` prop](#removed-breadcrumbs-deprecated-breadcrumbs-prop)
+  - [Removed `KonamiCode`](#removed-konami-code)
```
Comment (laurkim) [inline documentation/guides/migrating-from-v10-to-v11.md]:
```suggestion
  - [Removed `KonamiCode`](#removed-konamicode)
```

File: documentation/guides/migrating-from-v10-to-v11.md [hunk 5]
```diff
@@ -9,32 +9,34 @@ Polaris v11.0.0 ([full release notes](https://github.com/Shopify/polaris/release
...
+  - [Removed `DisplayText`](#removed-display-text)
```
Comment (laurkim) [inline documentation/guides/migrating-from-v10-to-v11.md]:
```suggestion
  - [Removed `DisplayText`](#removed-displaytext)
```

File: documentation/guides/migrating-from-v10-to-v11.md [hunk 6]
```diff
@@ -9,32 +9,34 @@ Polaris v11.0.0 ([full release notes](https://github.com/Shopify/polaris/release
...
+  - [Removed `Heading`](#removed-heading)
+  - [Removed `Subheading`](#removed-subheading)
+  - [Removed `Caption`](#removed-caption)
+  - [Removed `TextStyle`](#removed-text-style)
```
Comment (laurkim) [inline documentation/guides/migrating-from-v10-to-v11.md:20]:
```suggestion
  - [Removed `TextStyle`](#removed-textstyle)
```

File: documentation/guides/migrating-from-v10-to-v11.md [hunk 7]
```diff
@@ -9,32 +9,34 @@ Polaris v11.0.0 ([full release notes](https://github.com/Shopify/polaris/release
...
+  - [Removed `VisuallyHidden`](#removed-visually-hidden)
+  - [Migrated `Stack` to `LegacyStack`](#migrated-stack-to-legacy-stack)
+  - [Migrated `Card` to `LegacyCard`](#migrated-card-to-legacy-card)
+  - [Migrated `Filters` to `LegacyFilters`](#migrated-filters-to-legacy-filters)
+  - [Migrated `Tabs` to `LegacyTabs`](#migrated-tabs-to-legacy-tabs)
```
Comment (laurkim) [inline documentation/guides/migrating-from-v10-to-v11.md]:
```suggestion
  - [Removed `TextStyle`](#removed-textstyle)
  - [Removed `VisuallyHidden`](#removed-visuallyhidden)
  - [Migrated `Stack` to `LegacyStack`](#migrated-stack-to-legacystack)
  - [Migrated `Card` to `LegacyCard`](#migrated-card-to-legacycard)
  - [Migrated `Filters` to `LegacyFilters`](#migrated-filters-to-legacyfilters)
  - [Migrated `Tabs` to `LegacyTabs`](#migrated-tabs-to-legacytabs)
```

File: documentation/guides/migrating-from-v10-to-v11.md [hunk 8]
```diff
@@ -9,32 +9,34 @@ Polaris v11.0.0 ([full release notes](https://github.com/Shopify/polaris/release
...
+  - [Renamed `Inline`](#renamed-inline)
+  - [Renamed `AlphaStack`](#renamed-alpha-stack)
+  - [Renamed `AlphaCard`](#renamed-alpha-card)
+  - [Renamed `AlphaFilters`](#renamed-alpha-filters)
+  - [Renamed `AlphaTabs`](#renamed-alpha-tabs)
```
Comment (laurkim) [inline documentation/guides/migrating-from-v10-to-v11.md]:
```suggestion
  - [Renamed `AlphaStack`](#renamed-alphastack)
  - [Renamed `AlphaCard`](#renamed-alphacard)
  - [Renamed `AlphaFilters`](#renamed-alphafilters)
  - [Renamed `AlphaTabs`](#renamed-alphatabs)
```

File: documentation/guides/migrating-from-v10-to-v11.md [hunk 9]
```diff
@@ -52,236 +54,360 @@ Webpack version 4 is no longer supported. Webpack 5 is the minimum supported ver
...
 
-**Legacy TypeScript Support**
+### Legacy TypeScript Support
 
 Polaris no longer supports multiple versions of TypeScript with `downlevel-dts`. Polaris only builds one set of types based on the current version of TypeScript in the project.
 
 ## Components
 
-### Removed Collapsible deprecated preventMeasuringOnChildrenUpdate prop
+The following components have either been renamed, migrated, or removed. Please review each component section to determine whether you can resolve these breaking changes with a migration and/or if they must be handled manually.
```
Comment (laurkim) [inline documentation/guides/migrating-from-v10-to-v11.md]:
```suggestion
The following components have either been renamed, migrated, or removed. Please review each component section to determine whether you can resolve these breaking changes with a migration or if they need to updated manually.
```

File: documentation/guides/migrating-from-v10-to-v11.md [hunk 10]
```diff
@@ -52,236 +54,360 @@ Webpack version 4 is no longer supported. Webpack 5 is the minimum supported ver
...
 
-<br />
+To replace the six removed typography components (`DisplayText`, `Heading`, `Subheading`, `Caption`, `TextStyle`, and `VisuallyHidden`), you can run the [v10-react-replace-text-component](https://polaris.shopify.com/tools/polaris-migrator#v10-react-replace-text-components) migration. Please reference the [recommended component migration workflow](#recommended-component-migration-workflow) section below for additional migration support.
 
 ```sh
 npx @shopify/polaris-migrator v10-react-replace-text-components <path>
 ```
 
-<br />
+Note if you only want to run this migration on `DisplayText` and not the other legacy text components you can run:
```
Comment (laurkim) [inline documentation/guides/migrating-from-v10-to-v11.md]:
```suggestion
> **Note**: To migrate only the `DisplayText` component, the migration can be supplemented with the `--componentName='DisplayText'` flag.
```

File: documentation/guides/migrating-from-v10-to-v11.md [hunk 11]
```diff
@@ -52,236 +54,360 @@ Webpack version 4 is no longer supported. Webpack 5 is the minimum supported ver
...
 
-<br />
+To replace the six removed typography components (`DisplayText`, `Heading`, `Subheading`, `Caption`, `TextStyle`, and `VisuallyHidden`), you can run the [v10-react-replace-text-components](https://polaris.shopify.com/tools/polaris-migrator#v10-react-replace-text-components) migration. Please reference the [recommended component migration workflow](#recommended-component-migration-workflow) section below for additional migration support.
 
 ```sh
 npx @shopify/polaris-migrator v10-react-replace-text-components <path>
 ```
 
-<br />
+Note if you only want to run this migration on `Heading` and not the other legacy text components you can run:
```
Comment (laurkim) [inline documentation/guides/migrating-from-v10-to-v11.md]:
```suggestion
> **Note**: To migrate only the `Heading` component, the migration can be supplemented with the `--componentName='Heading'` flag.
```

File: documentation/guides/migrating-from-v10-to-v11.md [hunk 12]
```diff
@@ -52,236 +54,360 @@ Webpack version 4 is no longer supported. Webpack 5 is the minimum supported ver
...
 
-<br />
+To replace the six removed typography components (`DisplayText`, `Heading`, `Subheading`, `Caption`, `TextStyle`, and `VisuallyHidden`), you can run the [v10-react-replace-text-components](https://polaris.shopify.com/tools/polaris-migrator#v10-react-replace-text-components) migration. Please reference the [recommended component migration workflow](#recommended-component-migration-workflow) section below for additional migration support.
 
 ```sh
 npx @shopify/polaris-migrator v10-react-replace-text-components <path>
 ```
 
-<br />
+Note if you only want to run this migration on `Subheading` and not the other legacy text components you can run:
```
Comment (laurkim) [inline documentation/guides/migrating-from-v10-to-v11.md]:
```suggestion
> **Note**: To migrate only the `Subheading` component, the migration can be supplemented with the `--componentName='Subheading'` flag.
```

File: documentation/guides/migrating-from-v10-to-v11.md [hunk 13]
```diff
@@ -485,51 +691,104 @@ To rename `AlphaTabs` to `Tabs`, you can run the generic [react-rename-component
...
+<AlphaTab
+```
+
+```
+Polaris-AlphaTab
+```
+
+### Recommended component migration workflow
+
+When running component migrations we suggest following the following workflow:
```
Comment (aaronccasanova) [inline documentation/guides/migrating-from-v10-to-v11.md]:
```suggestion
When running component migrations we suggest the following workflow:
```

Light suggestion

File: documentation/guides/migrating-from-v10-to-v11.md [hunk 14]
```diff
@@ -851,3 +1178,37 @@ npx @shopify/polaris-migrator v11-styles-replace-custom-property-zindex <path>
 | `--p-z-10`       | `--p-z-index-10`  |
 | `--p-z-11`       | `--p-z-index-11`  |
 | `--p-z-12`       | `--p-z-index-12`  |
+
+### Recommended token migration workflow
+
+When running token migrations we suggest following the following workflow:
```
Comment (aaronccasanova) [inline documentation/guides/migrating-from-v10-to-v11.md]:
```suggestion
When running token migrations we suggest the following workflow:
```

Same as above

File: documentation/guides/migrating-from-v10-to-v11.md (hunk 3 above)
Comment (lgriffee) [reply documentation/guides/migrating-from-v10-to-v11.md]: Thanks so much for catching this Lo!! ❤️

File: documentation/guides/migrating-from-v10-to-v11.md (hunk 10 above)
Comment (lgriffee) [reply documentation/guides/migrating-from-v10-to-v11.md]: Love your suggestion here! Thanks Lo!

File: documentation/guides/migrating-from-v10-to-v11.md (hunk 13 above)
Comment (lgriffee) [reply documentation/guides/migrating-from-v10-to-v11.md]: Thanks for catching this Aaron!

File: documentation/guides/migrating-from-v10-to-v11.md [hunk 15]
```diff
@@ -52,236 +54,360 @@ Webpack version 4 is no longer supported. Webpack 5 is the minimum supported ver
...
 
-**Legacy TypeScript Support**
+### Legacy TypeScript Support
 
 Polaris no longer supports multiple versions of TypeScript with `downlevel-dts`. Polaris only builds one set of types based on the current version of TypeScript in the project.
 
 ## Components
 
-### Removed Collapsible deprecated preventMeasuringOnChildrenUpdate prop
+The following components have either been renamed, migrated, or removed. Please review each component section to determine whether you can resolve these breaking changes with a migration or if they need to updated manually.
```
Comment (chloerice) [inline documentation/guides/migrating-from-v10-to-v11.md]:
```suggestion
The following components have either been renamed, migrated, or removed. Please review each component section to determine whether you can resolve these breaking changes with a migration or if they need to be updated manually.
```

File: documentation/guides/migrating-from-v10-to-v11.md [hunk 16]
```diff
@@ -52,236 +54,360 @@ Webpack version 4 is no longer supported. Webpack 5 is the minimum supported ver
...
 
-To manually migrate `DisplayText` to `Text`, use the following methods:
+```sh
+npx @shopify/polaris-migrator v10-react-replace-text-components --componentName='DisplayText' <path>
+```
+
+#### Post-migration validation
 
-#### Small
+After migrating use the following to check for any additional instances of `DisplayText` across all file types:
```
Comment (chloerice) [inline documentation/guides/migrating-from-v10-to-v11.md]:
```suggestion
After migrating, search for React instances or CSS class overrides of the `DisplayText` component across all file types in your project:
```

File: documentation/guides/migrating-from-v10-to-v11.md [hunk 17]
```diff
@@ -52,236 +54,360 @@ Webpack version 4 is no longer supported. Webpack 5 is the minimum supported ver
...
+
+```
```
Comment (chloerice) [inline documentation/guides/migrating-from-v10-to-v11.md]:
```suggestion
```jsx
```

File: documentation/guides/migrating-from-v10-to-v11.md [hunk 18]
```diff
@@ -52,236 +54,360 @@ Webpack version 4 is no longer supported. Webpack 5 is the minimum supported ver
...
+<DisplayText
+```
+
+```
```
Comment (chloerice) [inline documentation/guides/migrating-from-v10-to-v11.md]:
```suggestion
```css
```

**Review by chloerice — APPROVED** [review]
This looks great Laura! 🙌🏽 Just suggested a typo fix and some clarifying text for the post-migration validation sections.

**Pull request merged**